### PR TITLE
feat(voter-dapp): Display no-votes in Resolved Requests appropriately, add all-in-one testing script

### LINKED ISSUE #1925: Voter dApp: Resolved Admin price requests display "INVALID ADMIN VOTE" for votes that user did not participate in
**Example**

Screen Shot 2020-08-31 at 10 20 25

### LINKED ISSUE #1926: Voter dApp: Create scripts to make testing a lot easier
**Description**

Current way to test voter dapp is to run a series of `.js` scripts, this is unscalable. Best to package them into one comprehensive `bash` script that guides user through testing.

### PR DESCRIPTION
**Pull request by nicholaspai**
**Motivation**

#1926 
#1925 

**Summary**

Script will run user through commit-reveal-claim cycle for normal and admin price requests, with and without a 2key contract. Script takes care of deploying contracts, submitting price requests, advancing round phases, and getting Metamask set up properly.

**Details**

Example log of script:
<details>
  <summary>Expand</summary>

```sh
➜  voter-dapp git:(npai/fix-voter-dapp) ✗ ./run_tests.sh

####################################################################
#                                                                  #
# 🐉 Starting Voter dApp tests 🐉                                  #
#                                                                  #
####################################################################


####################################################################
#                                                                  #
# 0/16. Prerequisites                                              #
#                                                                  #
####################################################################

- Have you started Ganache in a separate window at port 9545? Be sure to use the following mnemonic: "candy maple cake sugar pudding cream honey rich smooth crumble sweet treat" 
1) Yes
2) Help
3) Exit
#? 1

####################################################################
#                                                                  #
# 1/16. Deploying voting contracts on test network                 #
#                                                                  #
####################################################################

yarn run v1.22.0
$ /Users/nicholaspai/UMA/protocol/node_modules/.bin/truffle migrate --reset --network test

Compiling your contracts...
===========================
> Everything is up to date, there is nothing to compile.



Starting migrations...
======================
> Network name:    'test'
> Network id:      1598895416217
> Block gas limit: 9000000 (0x895440)


1_initial_migration.js
======================

   Replacing 'Migrations'
   ----------------------
   > transaction hash:    0x64ae596174c82832a6d95a619f8f74fae067d263d2f93cc0a88591e778a35e9c
   > Blocks: 0            Seconds: 0
   > contract address:    0x1FdE8F79B63cbACBefa2E7e3e158Bae2380A1ae9
   > block number:        155
   > block timestamp:     1598895841
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.90695244
   > gas used:            168502 (0x29236)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.00337004 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.00337004 ETH


2_deploy_finder.js
==================

   Replacing 'Finder'
   ------------------
   > transaction hash:    0xe98cdc6e0761c69af607a9ac3daa4d166b188aac865f1e4c27e8b3f1eb7cdccc
   > Blocks: 0            Seconds: 0
   > contract address:    0x48DD0dcaeCe8EbDe0A5F2E7A69B1082cd318cCd1
   > block number:        157
   > block timestamp:     1598895842
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.8994685
   > gas used:            331918 (0x5108e)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.00663836 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.00663836 ETH


3_deploy_timer.js
=================

   Replacing 'Timer'
   -----------------
   > transaction hash:    0xf8a256d98155e641fd7b87c67939595ded29c7dfa08e26a5377ed3f50690f1e5
   > Blocks: 0            Seconds: 0
   > contract address:    0x13C72f3c8bC7AfC2609a9da61a0d185c50085Cf8
   > block number:        159
   > block timestamp:     1598895842
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.89671044
   > gas used:            110624 (0x1b020)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.00221248 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.00221248 ETH


4_deploy_voting_token.js
========================

   Replacing 'VotingToken'
   -----------------------
   > transaction hash:    0xd3ee48b8834f9de7d82be00362773c2a4e5e63937c61c020a88ab09cade59dcb
   > Blocks: 0            Seconds: 0
   > contract address:    0x69797A4586692c4FafEA7b33F77aAc071Da0e2E9
   > block number:        161
   > block timestamp:     1598895843
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.8620625
   > gas used:            1705118 (0x1a049e)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.03410236 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.03410236 ETH


5_deploy_voting.js
==================

   Replacing 'IdentifierWhitelist'
   -------------------------------
   > transaction hash:    0xc04f30d1232ca790978b3006d5c68b119c7c8f14cc8143e7197a360b5f68b98b
   > Blocks: 0            Seconds: 0
   > contract address:    0x964839283Df5966A929dAD39c5ce5CdcaAf332E0
   > block number:        166
   > block timestamp:     1598895844
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.85186384
   > gas used:            341434 (0x535ba)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.00682868 ETH


   Replacing 'Voting'
   ------------------
   > transaction hash:    0x928bc04872dad678514d378332b820c7d5598877091c822908b4be0124af4f8b
   > Blocks: 0            Seconds: 0
   > contract address:    0x7419f8F92a40D72850267cE81Dc392ACd2540A0F
   > block number:        167
   > block timestamp:     1598895844
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.78436338
   > gas used:            3375023 (0x337faf)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.06750046 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.07432914 ETH


6_deploy_registry.js
====================

   Replacing 'Registry'
   --------------------
   > transaction hash:    0xf2d2fa6fc8af6172c45939ce609e6dbd216ba1c28019a694bbfe961919a25d0d
   > Blocks: 0            Seconds: 0
   > contract address:    0xe412Be7331B3452577c63569638354fe886FE405
   > block number:        172
   > block timestamp:     1598895845
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.75625148
   > gas used:            1239330 (0x12e922)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.0247866 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:           0.0247866 ETH


7_deploy_financial_contracts_admin.js
=====================================

   Replacing 'FinancialContractsAdmin'
   -----------------------------------
   > transaction hash:    0x95d476f706765d7529d2319337d09175278996c980ecf2eaaf3ad2965eadd024
   > Blocks: 0            Seconds: 0
   > contract address:    0x91eFe2c55d61fdf04c6409D5956Ff35eCb362d48
   > block number:        175
   > block timestamp:     1598895845
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.74835024
   > gas used:            322588 (0x4ec1c)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.00645176 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.00645176 ETH


8_deploy_store.js
=================

   Replacing 'Store'
   -----------------
   > transaction hash:    0x1a653e8dbe3b52c37503d826f938da0a784862607befeb58112ad409e2741415
   > Blocks: 0            Seconds: 0
   > contract address:    0x31840069353AacC84909Dc94427934DB660336be
   > block number:        178
   > block timestamp:     1598895846
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.714568
   > gas used:            1616458 (0x18aa4a)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.03232916 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.03232916 ETH


9_deploy_governor.js
====================

   Replacing 'Governor'
   --------------------
   > transaction hash:    0xafd4df71b2d9ee61399075c66a4876f67f7c2e1a1ee12d05fd6a92df4e455223
   > Blocks: 0            Seconds: 0
   > contract address:    0x7edf203fC69b33d7CaFa4837BD1a73bC1C58F618
   > block number:        181
   > block timestamp:     1598895847
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.67906758
   > gas used:            1702595 (0x19fac3)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.0340519 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:           0.0340519 ETH


10_deploy_designated_voting_factory.js
======================================

   Replacing 'DesignatedVotingFactory'
   -----------------------------------
   > transaction hash:    0x35fd109229bf05befdf736f34aada7849a5f1d9c6026a469d794d0dfa673759c
   > Blocks: 0            Seconds: 0
   > contract address:    0x4e3e408000e7b39950EbB0091aBaC542bee66179
   > block number:        186
   > block timestamp:     1598895848
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.62683426
   > gas used:            2425289 (0x2501c9)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.04850578 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.04850578 ETH


11_support_identifiers.js
=========================

   > Saving migration to chain.
   -------------------------------------
   > Total cost:                   0 ETH


12_deploy_testnet_token.js
==========================

   Replacing 'TestnetERC20'
   ------------------------
   > transaction hash:    0x21d41a9c9906d9e8b1749efbaf9be82a1c6dde121a699800f75ca6d5fd0d5b3f
   > Blocks: 0            Seconds: 0
   > contract address:    0x331E1356344CEdd99E28c27DEB5339166E916c59
   > block number:        206
   > block timestamp:     1598895850
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.59539576
   > gas used:            747216 (0xb66d0)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.01494432 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.01494432 ETH


13_deploy_tokenfactory.js
=========================

   Replacing 'TokenFactory'
   ------------------------
   > transaction hash:    0x86aef425fc274d65073524148def14c92ed652ea98d1017ebd7ab726e205930e
   > Blocks: 0            Seconds: 0
   > contract address:    0xb0100efa20b6CF655FD8233E3748657D17d521b4
   > block number:        208
   > block timestamp:     1598895850
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.55333068
   > gas used:            2075975 (0x1fad47)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.0415195 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:           0.0415195 ETH


14_deploy_expiring_multi_party_creator.js
=========================================

   Replacing 'AddressWhitelist'
   ----------------------------
   > transaction hash:    0x3b5611cb2653a9b29e84d219462da51703b1e40a26337170ab11cad27044daf0
   > Blocks: 0            Seconds: 0
   > contract address:    0x8cE2D79C5639B9CAA64394e3259b6d5CbfD6035c
   > block number:        210
   > block timestamp:     1598895851
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.54126926
   > gas used:            575792 (0x8c930)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.01151584 ETH


   Replacing 'ExpiringMultiPartyLib'
   ---------------------------------
   > transaction hash:    0xc637428c88df23b9e0108057e131327db257b20052965df2d2cc1e42b2d20336
   > Blocks: 0            Seconds: 0
   > contract address:    0x98fd7Bd970f4faB598d810d4194167cAdF3D3941
   > block number:        213
   > block timestamp:     1598895851
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.43396814
   > gas used:            5227334 (0x4fc346)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.10454668 ETH


   Linking
   -------
   * Contract: ExpiringMultiPartyCreator <--> Library: ExpiringMultiPartyLib (at address: 0x98fd7Bd970f4faB598d810d4194167cAdF3D3941)

   Replacing 'ExpiringMultiPartyCreator'
   -------------------------------------
   > transaction hash:    0x3a526a8181ee83aa7e13761e690d5dad43cb54d149589102d2b41e47ab420b96
   > Blocks: 0            Seconds: 0
   > contract address:    0xC678de088388d961a6CA623Ac7fC8b98d975918c
   > block number:        214
   > block timestamp:     1598895852
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.41628174
   > gas used:            884320 (0xd7e60)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.0176864 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.13374892 ETH


15_deploy_local_weth.js
=======================

   Replacing 'WETH9'
   -----------------
   > transaction hash:    0x47cba3138d6befa9bdd6b0b90fdff6d9408afa5dbcc0614143ecab7175353121
   > Blocks: 0            Seconds: 0
   > contract address:    0x9515d3C99AFa4CdF9a8CE6674e5ce1cFcb6eA88b
   > block number:        217
   > block timestamp:     1598895852
   > account:             0x627306090abaB3A6e1400e9345bC60c78a8BEf57
   > balance:             9999999998.4046689
   > gas used:            504749 (0x7b3ad)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.01009498 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.01009498 ETH


Summary
=======
> Total deployments:   17
> Final cost:          0.4670853 ETH


✨  Done in 16.48s.
- ✅ Migration complete!

####################################################################
#                                                                  #
# 2/16. Submitting price requests                                  #
#                                                                  #
####################################################################

yarn run v1.22.0
$ /Users/nicholaspai/UMA/protocol/node_modules/.bin/truffle exec ./scripts/local/RequestOraclePrice.js --network test --identifier USDETH-Commit+Reveal --time 1570000000
> Warning: possible unsupported (undocumented in help) command line option: --identifier,--time
Using network 'test'.

Creator Added: 0x627306090abaB3A6e1400e9345bC60c78a8BEf57
Registered contract: 0x627306090abaB3A6e1400e9345bC60c78a8BEf57
Price requested for USDETH-Commit+Reveal @ Wed Oct 02 2019 03:06:40 GMT-0400 (Eastern Daylight Time)
✨  Done in 5.93s.
yarn run v1.22.0
$ /Users/nicholaspai/UMA/protocol/node_modules/.bin/truffle exec ./scripts/local/RequestOraclePrice.js --network test --identifier USDETH-2key-Commit+Reveal --time 1570000000
> Warning: possible unsupported (undocumented in help) command line option: --identifier,--time
Using network 'test'.

Price requested for USDETH-2key-Commit+Reveal @ Wed Oct 02 2019 03:06:40 GMT-0400 (Eastern Daylight Time)
✨  Done in 4.53s.
yarn run v1.22.0
$ /Users/nicholaspai/UMA/protocol/node_modules/.bin/truffle exec ./scripts/local/ProposeAdmin.js --network test
Using network 'test'.

Admin proposal transaction 0x10a7e201544553542d4e45572d4944454e54494649455200000000000000000000000000
Transferred ownership of the IdentifierWhitelist to the Governor

  
      Newly Proposed DVM Identifier: 
  
      - TEST-NEW-IDENTIFIER (UTF8)
      - 0x544553542d4e45572d4944454e544946494552 (HEX)
  
    
✨  Done in 4.18s.
- ✅ Price requests submitted!

####################################################################
#                                                                  #
# 3/16. Advancing to start of next voting round                    #
#                                                                  #
####################################################################

yarn run v1.22.0
$ /Users/nicholaspai/UMA/protocol/node_modules/.bin/truffle exec ./scripts/local/AdvanceToNextVotingPhase.js --network test
Using network 'test'.

Moved from phase 1 to 0
✨  Done in 3.95s.
- ✅ Advanced to next commit phase!

####################################################################
#                                                                  #
# 4/16. Setting up Metamask                                        #
#                                                                  #
####################################################################

- ‼️ Before testing the voter dApp, you will need to import the account holding voter tokens into your Metamask browser extension. Type "Help" to see account details and "Continue" when you are ready
1) Continue
2) Help
3) Exit
#? 1 

####################################################################
#                                                                  #
# 5/16. Starting Voter dApp                                        #
#                                                                  #
####################################################################

- Start the voter dApp by switching to the packages/voter-dapp directory (cd ../voter-dapp) and starting the server on port 3000 by running "yarn start" from a different window, point the network to your local Ganache at port 9545, and then enter 1 to continue: 
1) Continue
2) Help
3) Exit
#? 1

####################################################################
#                                                                  #
# 6/16. Committing without the 2 Key wallet                        #
#                                                                  #
####################################################################

- Commit votes for [USDETH-Commit+Reveal, Admin-0]. Click Edit to enter a vote, and then Save to submit a transaction via Metamask. You can enter any price for the non-admin price request, and check that the admin Commit button displays a radio selection of YES and NO. Prior to committing a vote, the Current Vote column should only display an Edit button. After committing, your vote should be displayed (and translated into YES or NO for Admin votes)
1) Continue
2) Help
3) Exit
#? 1
- Advancing to the Reveal phase of the current voting round
yarn run v1.22.0
$ /Users/nicholaspai/UMA/protocol/node_modules/.bin/truffle exec ./scripts/local/AdvanceToNextVotingPhase.js --network test
Using network 'test'.

Moved from phase 0 to 1
✨  Done in 3.96s.

####################################################################
#                                                                  #
# 7/16. Snapshotting voting token balances                         #
#                                                                  #
####################################################################

- Before reveals can take place, a snapshot of current voting token balances must be taken
yarn run v1.22.0
$ /Users/nicholaspai/UMA/protocol/node_modules/.bin/truffle exec ./scripts/local/SnapshotCurrentRound.js --network test
Using network 'test'.

Snapshotted current round: [
  {
    logIndex: 0,
    transactionIndex: 0,
    transactionHash: '0xdfe8d5cd99278d58f4083e58f9e0421a78944d8521e6a1640db9f3f8b00f8ad1',
    blockHash: '0xef2c910c64e65ef428a2615710304f2cffe4001e45ef06ecb3ef5f8ff0465c87',
    blockNumber: 231,
    address: '0x69797A4586692c4FafEA7b33F77aAc071Da0e2E9',
    data: '0x0000000000000000000000000000000000000000000000000000000000000001',
    topics: [
      '0x8030e83b04d87bef53480e26263266d6ca66863aa8506aca6f2559d18aa1cb67'
    ],
    type: 'mined',
    id: 'log_76eb7a0e'
  }
]
✨  Done in 4.02s.

####################################################################
#                                                                  #
# 8/16. Revealing without the 2 Key wallet                         #
#                                                                  #
####################################################################

- Reveal votes for [USDETH-Commit+Reveal, Admin-0]. Click Edit to select the commits to reveal, and then Save to submit a transaction via Metamask. Prior to and after revealing, your committed values should be displayed (and translated into YES or NO for Admin votes). After revealing successfully, the Status column should display Revealed
1) Continue
2) Help
3) Exit
#? 1
- Advancing to the next Voting round
yarn run v1.22.0
$ /Users/nicholaspai/UMA/protocol/node_modules/.bin/truffle exec ./scripts/local/AdvanceToNextVotingPhase.js --network test
Using network 'test'.

Moved from phase 1 to 0
✨  Done in 4.05s.

####################################################################
#                                                                  #
# 9/16. Claiming rewards                                           #
#                                                                  #
####################################################################

- Claim rewards by clicking Claim Your Rewards under Retrieve Voting Rewards. You should receive your previous round's token balance (100,000,000) multiplied by the inflation rate (0.05%) multiplied by 2 for the two price requests (e,g, you should have received 100,000 tokens total). Note that you are the one and only voter so you should always have voted with the majority.
1) Continue
2) Help
3) Exit
#? 1

####################################################################
#                                                                  #
# 10/16. Resolved Requests                                         #
#                                                                  #
####################################################################

- Hit the Resolved Requests button and verify that your submitted votes are displayed properly. Hit continue once you've confirmed this
1) Continue
2) Help
3) Exit
#? 1

####################################################################
#                                                                  #
# 11/16. Create a voting proxy contract                            #
#                                                                  #
####################################################################

- Follow this tutorial to set up your 2Key wallet via the voter dApp: https://docs.umaproject.org/tutorials/voting-2key. Use your current address as the Cold Wallet Address and transfer your tokens to the proxy contract. You will be prompted to connect again to the voter dApp, but if this doesn't happen automatically then you can simply refresh the page. The Resolved Requests will no longer show that you voted on the two resolved price requests because your voting account has changed to the proxy contract's address
1) Continue
2) Help
3) Exit
#? 1

####################################################################
#                                                                  #
# 12/16. Committing with the 2 Key wallet                          #
#                                                                  #
####################################################################

- Commit votes for [USDETH-2key-Commit+Reveal].
1) Continue
2) Help
3) Exit
#? 1
- Advancing to the Reveal phase of the current voting round
yarn run v1.22.0
$ /Users/nicholaspai/UMA/protocol/node_modules/.bin/truffle exec ./scripts/local/AdvanceToNextVotingPhase.js --network test
Using network 'test'.

Moved from phase 0 to 1
✨  Done in 4.21s.

####################################################################
#                                                                  #
# 13/16. Snapshotting voting token balances                        #
#                                                                  #
####################################################################

- Before reveals can take place, a snapshot of current voting token balances must be taken
yarn run v1.22.0
$ /Users/nicholaspai/UMA/protocol/node_modules/.bin/truffle exec ./scripts/local/SnapshotCurrentRound.js --network test
Using network 'test'.

Snapshotted current round: [
  {
    logIndex: 0,
    transactionIndex: 0,
    transactionHash: '0x72ac4474aa5baf9238d75f1c838ad9be2ed74834882b0181048715c4dd7d690c',
    blockHash: '0x284c88b10da070c619a21998730e8b8643fa9f0d9a40ba6778265a5a3d2b6a6a',
    blockNumber: 239,
    address: '0x69797A4586692c4FafEA7b33F77aAc071Da0e2E9',
    data: '0x0000000000000000000000000000000000000000000000000000000000000002',
    topics: [
      '0x8030e83b04d87bef53480e26263266d6ca66863aa8506aca6f2559d18aa1cb67'
    ],
    type: 'mined',
    id: 'log_e3a94705'
  }
]
✨  Done in 3.97s.

####################################################################
#                                                                  #
# 14/16. Revealing with the 2 Key wallet                           #
#                                                                  #
####################################################################

- Reveal votes for [USDETH-2key-Commit+Reveal, Admin-0].
1) Continue
2) Help
3) Exit
#? 1
- Advancing to the next Voting round
yarn run v1.22.0
$ /Users/nicholaspai/UMA/protocol/node_modules/.bin/truffle exec ./scripts/local/AdvanceToNextVotingPhase.js --network test
Using network 'test'.

Moved from phase 1 to 0
✨  Done in 4.09s.

####################################################################
#                                                                  #
# 15/16. Claiming rewards                                          #
#                                                                  #
####################################################################

- Claim rewards by clicking Claim Your Rewards under Retrieve Voting Rewards. You should receive your previous round's token balance (100,100,000) multiplied by the inflation rate (0.05%) (e,g, you should have received 50,050 new tokens). Note that you are the one and only voter so you should always have voted with the majority.
1) Continue
2) Help
3) Exit
#? 1

####################################################################
#                                                                  #
# 16/16. Resolved Requests                                         #
#                                                                  #
####################################################################

- Hit the Resolved Requests button and verify that your submitted votes are displayed properly. Hit continue once you've confirmed this
1) Continue
2) Help
3) Exit
#? 1
🕺🏾💃🏻 Test complete!

```
</details>

**Issue(s)**

Fixes #1925 
Fixes #1926

### CODE REVIEW

File: packages/voter-dapp/README.md [hunk 1]
```diff
@@ -22,42 +22,9 @@ The source code for this React app can be found in `./src`. At a minimum, you mu
...
-1. To commit a vote, click "Edit", enter a number, check the box in the column to the right of the price feed identifier, and click "Save" below the "Active Requests" dashboard. You can commit several votes at once. Clicking "Save" will prompt you to sign and submit a [`batchCommit` transaction ](https://docs.umaproject.org/uma/contracts/VotingInterface.html#VotingInterface-batchCommit-struct-VotingInterface-Commitment---).
-1. Once you are done committing votes, advance to the reveal phase in your other console tab: `yarn truffle exec ./scripts/local/AdvanceToNextVotingPhase.js --network test`
-1. Either refresh the page or wait for the dApp to prompt you to sign another message. You will need to sign a message at the beginning of each voting phase. Now, you should see an option to reveal your committed vote.
-1. To reveal your vote, check the checkbox and click "Reveal Selected". This will prompt you to sign and submit a [`batchReveal` transaction](https://docs.umaproject.org/uma/contracts/VotingInterface.html#VotingInterface-batchReveal-struct-VotingInterface-Reveal---)
-1. If your vote was successfully revealed, the "Status" column should switch to "Revealed"
+### Testing
```
Comment (chrismaree) [inline packages/voter-dapp/README.md:25]: ooo I like this

File: packages/voter-dapp/run_tests.sh [hunk 2]
```diff
@@ -1,8 +1,256 @@
...
+echo -e '\n####################################################################'
+echo '#                                                                  #'
+echo '# 4/16. Setting up Metamask                                        #'
+echo '#                                                                  #'
+echo -e '####################################################################\n'
+echo "- ‼️ Before testing the voter dApp, you will need to import the account holding voter tokens into your Metamask browser extension. Type \"Help\" to see account details and \"Continue\" when you are ready"
+select yn in "Continue" "Help" "Exit"; do
+    case $yn in
+        Continue ) break;;
+        Help ) echo "🚸 Import the following Private Key into Metamask: https://metamask.zendesk.com/hc/en-us/articles/360015489331-Importing-an-Account" && yarn truffle exec ./scripts/local/getPrivateKeyFromMnemonic.js --mnemonic "candy maple cake sugar pudding cream honey rich smooth crumble sweet treat" --network test;;
```
Comment (chrismaree) [inline packages/voter-dapp/run_tests.sh:75]: this is very cool.

File: packages/voter-dapp/src/ResolvedRequests.js [hunk 3]
```diff
@@ -250,13 +250,15 @@ function ResolvedRequests({ votingAccount }) {
                 <TableCell>{hexToUtf8(resolutionData.identifier)}</TableCell>
                 <TableCell>{formatDate(resolutionData.time, web3)}</TableCell>
                 <TableCell>Resolved</TableCell>
-                <TableCell>{isAdminVote ? translateAdminVote(userVote) : userVote}</TableCell>
+                <TableCell>{isAdminVote && userVote !== "No Vote" ? translateAdminVote(userVote) : userVote}</TableCell>
```
Comment (chrismaree) [inline packages/voter-dapp/src/ResolvedRequests.js:253]: why this change?

**Review by chrismaree — APPROVED** [review]
This is super awesome. I really like how you've been able to re-use existing scripts in your bash script. Great work @nicholaspai

File: packages/voter-dapp/src/ResolvedRequests.js (hunk 3 above)
Comment (nicholaspai) [reply packages/voter-dapp/src/ResolvedRequests.js:253]: Becuase otherwise an Admin vote will call `translateAdminVote("No Vote")` which returns "INVALID ADMIN VOTE". This is the easiest fix for that
